js/angular-material-datetimepicker.js

```diff
@@ -1,998 +1,1012 @@
+                var pr = parseInt(css(pickerEl, 'paddingRight').replace('px', '')) || 0;
+                var ml = parseInt(css(clock, 'marginLeft').replace('px', '')) || 0;
+                var mr = parseInt(css(clock, 'marginRight').replace('px', '')) || 0;
+                //set width
+                var clockWidth = (w - (ml + mr + pl + pr));
+                clock.css('width', (clockWidth) + 'px');
+
+                var pL = parseInt(css(pickerEl, 'paddingLeft').replace('px', '')) || 0;
+                var pT = parseInt(css(pickerEl, 'paddingTop').replace('px', '')) || 0;
+                var mL = parseInt(css(clock, 'marginLeft').replace('px', '')) || 0;
+                var mT = parseInt(css(clock, 'marginTop').replace('px', '')) || 0;
+
+                var r = (clockWidth / 2);
+                var j = r / 1.2; //???
+
+                var points = [];
+
+                for (var h = 0; h < 12; ++h) {
+                  var x = j * Math.sin(Math.PI * 2 * (h / 12));
+                  var y = j * Math.cos(Math.PI * 2 * (h / 12));
+                  var left = (r + x + pL / 2) - (pL + mL);
+                  var top = (r - y - mT / 2) - (pT + mT);
+
+                  var hour = {
+                    left: left,
+                    top: top,
+                    value: (minuteMode ? (h * 5) : h), //5 for minute 60/12
+                    style: { 'margin-left': left + 'px', 'margin-top': top + 'px' }
+                  };
+
+                  if (minuteMode) {
+                    hour.display = hour.value < 10 ? ('0' + hour.value) : hour.value;
+                  } else {
 
-        return {
-          restrict: 'E',
-          template: template,
-          link: function (scope, element, attrs) {
-            var minuteMode = attrs.mode === 'minutes';
-            var picker = scope.picker;
-            //banking on the fact that there will only be one at a time
-            var componentRoot = document.querySelector('md-dialog.dtp');
-            var exec = function () {
-              var clock = angular.element(element[0].querySelector('.dtp-picker-clock')),
-                pickerEl = angular.element(componentRoot.querySelector('.dtp-picker'));
-
-              var w = componentRoot.querySelector('.dtp-content').offsetWidth;
-              var pl = parseInt(css(pickerEl, 'paddingLeft').replace('px', '')) || 0;
-              var pr = parseInt(css(pickerEl, 'paddingRight').replace('px', '')) || 0;
-              var ml = parseInt(css(clock, 'marginLeft').replace('px', '')) || 0;
-              var mr = parseInt(css(clock, 'marginRight').replace('px', '')) || 0;
-              //set width
-              var clockWidth = (w - (ml + mr + pl + pr));
-              clock.css('width', (clockWidth) + 'px');
-
-              var pL = parseInt(css(pickerEl, 'paddingLeft').replace('px', '')) || 0;
-              var pT = parseInt(css(pickerEl, 'paddingTop').replace('px', '')) || 0;
-            
```

Comment:
you should add the below code :

```
return moduleName;
```

It permits to get the module name with the return of require/import, it is useful when we inject ng-material-datetimepicker into our application